flash/core/classification.py

```diff
@@ -32,6 +35,79 @@ def after_uncollate(self, samples: Any) -> Any:
 
 class ClassificationTask(Task):
 
+    def __init__(
+        self,
+        num_classes: int,
+        model: Optional[torch.nn.Module] = None,
+        loss_fn: Optional[Callable] = None,
+        multilabel: bool = False,
+        optimizer: Type[torch.optim.Optimizer] = torch.optim.SGD,
+        metrics: Union[pl.metrics.Metric, Mapping, Sequence, None] = None,
+        learning_rate: float = 1e-3,
+    ):
+
+        super().__init__(
+            model=model, loss_fn=loss_fn, optimizer=optimizer, metrics=metrics, learning_rate=learning_rate
+        )
+
+        self.num_classes = num_classes
+        self.multilabel = multilabel
+
+        if isinstance(self.loss_fn, Mapping) and not self.loss_fn:
+            self.loss_fn = self.default_loss_fn
+
+    def step(self, batch: Sequence[torch.Tensor], batch_idx: int) -> torch.Tensor:
+        """
+        The training/validation/test step. Override for custom behavior.
+        """
+        x, y = batch
+        if self.multilabel:
+            y = y.float()
+        y_hat = self.forward(x)
+        output = {"y_hat": self.data_pipeline.before_uncollate(y_hat)}
+        losses = {name: l_fn(y_hat, y) for name, l_fn in self.loss_fn.items()}
+        logs = {}
+        for name, metric in self.metrics.items():
+            if isinstance(metric, pl.metrics.Metric):
+                metric(output["y_hat"], y.long())
+                logs[name] = metric  # log the metric itself if it is of type Metric
+            else:
+                logs[name] = metric(y_hat, y)
+        logs.update(losses)
+        if len(losses.values()) > 1:
+            logs["total_loss"] = sum(losses.values())
+            return logs["total_loss"], logs
+        output["loss"] = list(losses.values())[0]
+        output["logs"] = logs
+        output["y"] = y
+        return output
+
+    @predict_context
+    def predict(
+        self,
+        x: Any,
+        batch_idx: Optional[int] = None,
+        skip_collate_fn: bool = False,
+        dataloader_idx: Optional[int] = None,
+        data_pipeline: Optional[ClassificationDataPipeline] = None,
+    ) -> Any:    
+        if self.multilabel:
+            bind_method(self.data_pipeline, 
```

Comment:
I think this breaks if someone provides a data_pipeline as an argument but it works on the default pipeline.